### PR TITLE
Assignment - 2

### PR DESCRIPTION
**Pull request by priyanka0901**

### CODE REVIEW

File: src/routes/calculator/modules/calculator.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+export const calcState = {
+    calValue : 0,
+    calBtns : ["1", "2", "3", "C ","4", "5", "6", "=", "7", "8", "9", ".", "+", "0", "-", "*", "/"]
+}
+
+// reducer
+export const calcReducer = (state = calcState, action ) =>{
+    switch(action.type){
+        case 'ADD_ELEM':
```
Comment (Swathi8) [inline src/routes/calculator/modules/calculator.js]: @priyanka0901 This action type is confusing, `ADD_ELEM` is actually concatinating the string, but it sounds like it is for `addition`.

File: src/containers/AppContainer.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import { browserHistory, Router } from 'react-router'
 import { Provider } from 'react-redux'
+import {createStore} from 'redux';
```
Comment (Swathi8) [inline src/containers/AppContainer.js:5]: Why is this needed here?

File: src/routes/calculator/components/Calculator.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+ import React from 'react';
+ import PropTypes from 'prop-types';
+ import './calculator.scss';
+
+
+class Calculator  extends React.Component {
+    constructor(props){
```
Comment (Swathi8) [inline src/routes/calculator/components/Calculator.js:7]: I don't think constructor is needed here.

File: src/routes/calculator/components/Calculator.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        return (
+             <div className="App">
+               <div className="value-container">
+                 <input type="text" value={calcState.calValue} />
+               </div>
+               <div className="buttons-container">
+                    {calcState.calBtns.map((item, key) => {
+                        if(item == "C"){
+                            return(
+                               <button onClick={clear.bind(this) } key={key}>{item}</button>
```
Comment (Swathi8) [inline src/routes/calculator/components/Calculator.js:22]: You don't have to bind this(everywhere in this render) I think, try removing it and see

File: src/routes/calculator/components/Calculator.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    render() {
+        const { calcState, addElem, clear, equal } = this.props;
+        return (
+             <div className="App">
+               <div className="value-container">
+                 <input type="text" value={calcState.calValue} />
+               </div>
+               <div className="buttons-container">
+                    {calcState.calBtns.map((item, key) => {
+                        if(item == "C"){
```
Comment (Swathi8) [inline src/routes/calculator/components/Calculator.js:20]: Use `===` always for comparision

File: src/routes/calculator/components/Calculator.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+             </div>
+           );
+    }
+}
+
+
+Calculator.PropTypes = {
+    calValue    : PropTypes.number.isRequired,
+    calBtns     : PropTypes.array.isRequired,
+    calReducer  : PropTypes.func.isRequired
```
Comment (Swathi8) [inline src/routes/calculator/components/Calculator.js:44]: Why are you checking for `calReducer` here?

File: src/routes/calculator/index.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { injectReducer } from '../../store/reducers';
+// import Calculator from './components/Calculator';
```
Comment (Swathi8) [inline src/routes/calculator/index.js:2]: Can you remove unecessary comments

File: src/routes/result/components/Result.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class Result extends React.Component {
+    constructor(props){
```
Comment (Swathi8) [inline src/routes/result/components/Result.js:5]: Don't use constructor if it is not needed

**Review by Swathi8 — CHANGES_REQUESTED** [review]

**Comment by Swathi8** [comment]
@priyanka0901 Found two errors while testing the app, I uploaded the video [here](https://drive.google.com/file/d/0Bwhr5B_lk7LtRFg3c3ktaUtnOEE/view)

File: src/routes/result/components/Result.js (hunk 8 above)
Comment (priyanka0901) [reply src/routes/result/components/Result.js:5]: Okay, I will verify all these mistakes.
